### PR TITLE
docs: Update ExtendedSplashScreen documentation

### PR DESCRIPTION
**Pull request by lukeblevins**
GitHub Issue (If applicable): #574 

## PR Type

What kind of change does this PR introduce?

- Documentation content changes

## PR Checklist

Please check if your PR fulfills the following requirements:
- [ ] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Tested the changes where applicable:
	- [ ] UWP
	- [ ] WinUI
	- [ ] iOS
	- [ ] Android
	- [ ] WASM
	- [ ] MacOS
- [x] Updated the documentation as needed:
	- [ ] [General Doc Update](https://github.com/unoplatform/uno.toolkit.ui/tree/main/doc)
	- [ ] [Controls Doc Update](https://github.com/unoplatform/uno.toolkit.ui/tree/main/doc/controls)
	- [ ] [Extensions Doc Update](https://github.com/unoplatform/uno.toolkit.ui/tree/main/doc/helpers)
	- [ ] [controls-styles.md](https://github.com/unoplatform/uno.toolkit.ui/blob/main/doc/controls-styles.md)
	- [ ] [lightweight-styling.md (LightWeight Styling Resource Keys)](https://github.com/unoplatform/uno.toolkit.ui/blob/main/doc/lightweight-styling.md)
- [ ] [Runtime Tests and/or UI Tests](https://platform.uno/docs/articles/contributing/guidelines/creating-tests.html) for the changes have been added (for bug fixes / features) (if applicable)
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal)
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: doc/controls/ExtendedSplashScreen.md [hunk 1]
```diff
@@ -3,30 +3,60 @@ uid: Toolkit.Controls.ExtendedSplashScreen
 ---
 # ExtendedSplashScreen
 
-Represents a control, derived from [`LoadingView`](xref:Toolkit.Controls.LoadingView) that continues to display the native splash screen for an extended amount of time, allowing the application to fully load the UI. See [`LoadingView`](xref:Toolkit.Controls.LoadingView) for Properties and other Members that are inherited.
+Represents a control, derived from [`LoadingView`](xref:Toolkit.Controls.LoadingView) that displays a view which replicates the appearance and behavior of the target platform's native application loading screen.
```
Comment (agneszitte) [inline doc/controls/ExtendedSplashScreen.md:6]:
```suggestion
Represents a control, derived from [`LoadingView`](xref:Toolkit.Controls.LoadingView) that displays a view that replicates the appearance and behavior of the target platform's native application loading screen.
```

File: doc/controls/ExtendedSplashScreen.md [hunk 2]
```diff
@@ -3,30 +3,60 @@ uid: Toolkit.Controls.ExtendedSplashScreen
...
+
+Refer to [`LoadingView`](xref:Toolkit.Controls.LoadingView) for a list of inherited members.
+
+## Remarks
+
+The term _splash screen_ commonly refers to a view that is displayed when the user launches the application. Note that only a subset of the supported targets define a native splash screen construct that is displayed by the platform.
```
Comment (agneszitte) [inline doc/controls/ExtendedSplashScreen.md:12]:
```suggestion
The term _splash screen_ commonly refers to a view that is displayed when the user launches the application. Note that only a subset of the supported targets defines a native splash screen construct that is displayed by the platform.
```

File: doc/controls/ExtendedSplashScreen.md [hunk 3]
```diff
@@ -3,30 +3,60 @@ uid: Toolkit.Controls.ExtendedSplashScreen
...
-| `SplashIsEnabled` | `bool`                 | Gets a value that determines whether native splash is to be displayed for the current platform at runtime. Note for Android 12+ the native splash screen is always disabled irrespective of the Platforms value.                                     |
+| Property | Type | Description | Value |
+|----------|------|-------------|---------------|
+| **Platforms** | `SplashScreenPlatform` | Gets or sets the platform(s) where extended splash screen should be used. This is a flag enumeration, which allows for combining multiple values. | Default value is **All**. Other possible values include **Android**, **iOS**, **Windows**, **WebAssembly**, **Skia**, and **None**. |
+| **SplashIsEnabled** | `bool` | Gets a value representing whether the current environment is to display this splash screen. | **True** if the current platform is included in the **Platforms** property, otherwise **false**. |
+<!-- | **Window** | `Window` | Gets or sets a window for the application that is used to size the splash screen. | -->
```
Comment (agneszitte) [inline doc/controls/ExtendedSplashScreen.md:27]: This one is commented, why?

File: doc/controls/ExtendedSplashScreen.md [hunk 4]
```diff
@@ -3,30 +3,60 @@ uid: Toolkit.Controls.ExtendedSplashScreen
...
+
+## Methods
+
+| Method | Return Type | Description |
+|--------|-------------|-------------|
+| **Init** | `void` | Initializes the splash screen for the provided `Activity` instance. This static method should be invoked from the **OnCreate** override in `MainActivity`.<br/>**Note: This method only needs to be called on Android** |
 
 ## Usage
 
+### SplashScreenPlatform
```
Comment (agneszitte) [inline doc/controls/ExtendedSplashScreen.md:37]: Not so sure on this title here

File: doc/controls/ExtendedSplashScreen.md [hunk 5]
```diff
@@ -3,30 +3,60 @@ uid: Toolkit.Controls.ExtendedSplashScreen
...
+The following code snippet will only display the splash screen on Android and iOS.
+
 ```xml
-xmlns:utu="using:Uno.Toolkit.UI"
+<utu:ExtendedSplashScreen x:Name="Splash"
+                          Platforms="Android,iOS"
 ...
+```
+
+### ExtendedSplashScreen in XAML
```
Comment (agneszitte) [inline doc/controls/ExtendedSplashScreen.md:47]: should the title mention usage or LoadingContentTemplate instead here ?

File: doc/controls/ExtendedSplashScreen.md (hunk 4 above)
Comment (lukeblevins) [reply doc/controls/ExtendedSplashScreen.md]: Addressed this in recent commit.

File: doc/controls/ExtendedSplashScreen.md (hunk 3 above)
Comment (lukeblevins) [reply doc/controls/ExtendedSplashScreen.md:27]: @agneszitte @kazo0 The Window property does not appear to be set or used very often internally, so I was confused as to whether we should document it.

**Comment by xperiandri** [comment]
Will the fix to my remarks about the need for the splash screen in the template go to this PR or the other?

File: doc/controls/ExtendedSplashScreen.md [hunk 6]
```diff
@@ -3,30 +3,60 @@ uid: Toolkit.Controls.ExtendedSplashScreen
...
-xmlns:utu="using:Uno.Toolkit.UI"
+<utu:ExtendedSplashScreen x:Name="Splash"
+                          Platforms="Android,iOS"
 ...
+```
+
+### Set loading state content
+
+The example below demonstrates a typical use of `ExtendedSplashScreen` in XAML. The [**LoadingContentTemplate**](xref:Toolkit.Controls.LoadingView) property below is inherited from `LoadingView`. This property is used to define content that will be displayed during the loading/waiting state.
+
```
Comment (pictos) [inline doc/controls/ExtendedSplashScreen.md:50]: Not sure about it, but shouldn't we add `the` before `content`?
```suggestion
The example below demonstrates a typical use of `ExtendedSplashScreen` in XAML. The [**LoadingContentTemplate**](xref:Toolkit.Controls.LoadingView) property below is inherited from `LoadingView`. This property is used to define the content that will be displayed during the loading/waiting state.

```

File: doc/controls/ExtendedSplashScreen.md [hunk 7]
```diff
@@ -3,30 +3,60 @@ uid: Toolkit.Controls.ExtendedSplashScreen
...
-                            HorizontalAlignment="Stretch"
-                            VerticalAlignment="Stretch"
-                            HorizontalContentAlignment="Stretch"
-                            VerticalContentAlignment="Stretch"
-                            Platforms="Android,iOS,Windows">
+                          Platforms="Android,iOS"
+                          HorizontalAlignment="Stretch"
+                          VerticalAlignment="Stretch"
+                          HorizontalContentAlignment="Stretch"
+                          VerticalContentAlignment="Stretch">
```
Comment (pictos) [inline doc/controls/ExtendedSplashScreen.md:59]: `Stretch` isn't the default value for those properties?

**Review by pictos — APPROVED** [review]
Looks good, I've just two comments

File: doc/controls/ExtendedSplashScreen.md (hunk 6 above)
Comment (pedrojesus-work) [reply doc/controls/ExtendedSplashScreen.md:50]: Not sure about it, but shouldn't we add `the` before `content`?
```suggestion
The example below demonstrates a typical use of `ExtendedSplashScreen` in XAML. The [**LoadingContentTemplate**](xref:Toolkit.Controls.LoadingView) property below is inherited from `LoadingView`. This property is used to define the content that will be displayed during the loading/waiting state.

```

File: doc/controls/ExtendedSplashScreen.md (hunk 7 above)
Comment (pedrojesus-work) [reply doc/controls/ExtendedSplashScreen.md:59]: `Stretch` isn't the default value for those properties?

**Review by pedrojesus-work — APPROVED** [review]
Looks good, I've just two comments

**Comment by agneszitte** [comment]
> From [xperiandri](https://github.com/xperiandri):
> Will the fix to my remarks about the need for the splash screen in the template go to this PR or the other?

cc @jeromelaban, @nickrandolph

File: doc/controls/ExtendedSplashScreen.md [hunk 8]
```diff
@@ -1,48 +1,74 @@
...
+> [!TIP]
+> Users expect applications to load quickly after they launch them, so it is important to minimize the duration that a splash screen is displayed. `ExtendedSplashScreen` is intended to support a consistent user experience across platforms where scenarios necessitate a longer startup process.
+
+## Properties
+
+| Property | Type | Description | Value |
+|----------|------|-------------|---------------|
+| **Platforms** | `SplashScreenPlatform` | Gets or sets the platform(s) where extended splash screen should be used. This is a flag enumeration, which allows for combining multiple values. | Default value is **All**. Other possible values include **Android**, **iOS**, **Windows**, **WebAssembly**, **Skia**, and **None**. |
+| **SplashIsEnabled** | `bool` | Gets a value representing whether the current environment is to display this splash screen. | **True** if the current platform is included in the **Platforms** property, otherwise **false**. |
+<!-- | **Window** | `Window` | Gets or sets a window for the application that is used to size the splash screen. | -->
```
Comment (Xiaoy312) [inline doc/controls/ExtendedSplashScreen.md:27]:
```suggestion
<!-- | **Property** | `Type` | Description | Value -->
```

File: doc/controls/ExtendedSplashScreen.md [hunk 9]
```diff
@@ -1,48 +1,74 @@
...
+`ExtendedSplashScreen` is a custom implementation, and does not require the presence of a native splash screen. As such, it can serve as an application's standalone loading screen, regardless of whether the platform provides a native equivalent.
+
+> [!TIP]
+> Users expect applications to load quickly after they launch them, so it is important to minimize the duration that a splash screen is displayed. `ExtendedSplashScreen` is intended to support a consistent user experience across platforms where scenarios necessitate a longer startup process.
+
+## Properties
+
+| Property | Type | Description | Value |
+|----------|------|-------------|---------------|
+| **Platforms** | `SplashScreenPlatform` | Gets or sets the platform(s) where extended splash screen should be used. This is a flag enumeration, which allows for combining multiple values. | Default value is **All**. Other possible values include **Android**, **iOS**, **Windows**, **WebAssembly**, **Skia**, and **None**. |
```
Comment (Xiaoy312) [inline doc/controls/ExtendedSplashScreen.md:25]: keep the old text about how to specify multiple enum flags `"Android,iOS"`

File: doc/controls/ExtendedSplashScreen.md [hunk 10]
```diff
@@ -1,48 +1,74 @@
...
+## Usage
+
+### Specify platforms to display splash screen
+
+The following code snippet will only display the splash screen on Android and iOS by specifying a `SplashScreenPlatform` value for the `Platforms` property.
+
+```xml
+<utu:ExtendedSplashScreen x:Name="Splash"
+                          Platforms="Android,iOS"
+...
```
Comment (Xiaoy312) [inline doc/controls/ExtendedSplashScreen.md:44]:
```suggestion
<utu:ExtendedSplashScreen x:Name="Splash"
                          Platforms="Android,iOS" />
```

**Review by Xiaoy312 — APPROVED** [review]

File: doc/controls/ExtendedSplashScreen.md [hunk 11]
```diff
@@ -1,48 +1,74 @@
...
+                <ProgressRing IsActive="True"
+                              Grid.Row="1"
+                              VerticalAlignment="Center"
+                              HorizontalAlignment="Center"
+                              Height="100"
+                              Width="100" />
+            </Grid>
+        </DataTemplate>
+    </utu:ExtendedSplashScreen.LoadingContentTemplate>
+</utu:ExtendedSplashScreen>
```
[the remark this answers is not in the document]
Comment (kazo0) [reply doc/controls/ExtendedSplashScreen.md:73]: Perhaps pointing them to a sample app if we have one? If not, we should create one

[the remark this answers is not in the document]
Comment (xperiandri) [reply doc/controls/ExtendedSplashScreen.md:73]: Yes!

[the remark this answers is not in the document]
Comment (kazo0) [reply doc/controls/ExtendedSplashScreen.md:73]: @lukeblevins we should look to get this in before the release

File: doc/controls/ExtendedSplashScreen.md (hunk 3 above)
Comment (kazo0) [reply doc/controls/ExtendedSplashScreen.md]: @nickrandolph Is the Window property needed?

File: doc/controls/ExtendedSplashScreen.md [hunk 12]
```diff
@@ -3,50 +3,121 @@ uid: Toolkit.Controls.ExtendedSplashScreen
...
-                            VerticalContentAlignment="Stretch"
-                            Platforms="Android,iOS,Windows">
-    <utu:ExtendedSplashScreen.LoadingContentTemplate>
-        <DataTemplate>
+                          Platforms="Android,iOS" />
+```
+
+### Setup the splash screen
+
+The following code snippet demonstrate a suggested pattern for using the `ExtendedSplashScreen`. The first step is to define a custom `UserControl` that will be used as the main shell for the application content. This control will be used to host the `ExtendedSplashScreen` and the main application content.
```
Comment (agneszitte) [inline doc/controls/ExtendedSplashScreen.md:47]:
```suggestion
The following code snippet demonstrates a suggested pattern for using the `ExtendedSplashScreen`. The first step is to define a custom `UserControl` that will be used as the main shell for the application content. This control will be used to host the `ExtendedSplashScreen` and the main application content.
```

File: doc/controls/ExtendedSplashScreen.md [hunk 13]
```diff
@@ -3,50 +3,121 @@ uid: Toolkit.Controls.ExtendedSplashScreen
...
+    if (shell.RootFrame.Content == null)
+    {
+        shell.RootFrame.Navigate(typeof(MainPage), args.Arguments);
+    }
+
+    MainWindow.Activate();
+}
 ```
 
+With these changes, the splash screen will be displayed while the application first launches, and the main application content will be displayed once the loading state is complete.
```
Comment (agneszitte) [inline doc/controls/ExtendedSplashScreen.md:117]:
```suggestion
With these changes, the splash screen will be displayed when the application first launches and the main application content will be displayed once the loading state is complete.
```

File: doc/controls/ExtendedSplashScreen.md [hunk 14]
```diff
@@ -3,50 +3,121 @@ uid: Toolkit.Controls.ExtendedSplashScreen
...
+
+In order to prolong the splash screen display, you can use set the `Source` property of the `ExtendedSplashScreen` control to any custom implementation of the `ILoadable` interface. More information on how to use the `ILoadable` interface can be found in the [`LoadingView`](xref:Toolkit.Controls.LoadingView#iloadable) documentation.
```
Comment (agneszitte) [inline doc/controls/ExtendedSplashScreen.md:119]:
```suggestion
In order to prolong the splash screen display, you can set the `Source` property of the `ExtendedSplashScreen` control to any custom implementation of the `ILoadable` interface. More information on how to use the `ILoadable` interface can be found in the [`LoadingView`](xref:Toolkit.Controls.LoadingView#iloadable) documentation.
```

**Comment by kazo0** [comment]
I'm wondering if we need to cover the difference in setup if you are using the Navigation extensions? using a ShellModel that navigates to your MainModel when it starts?

**Review by kazo0 — APPROVED** [review]

**Pull request merged**